gui/syncthing/core/validDeviceidDirective.js

```diff
@@ -15,6 +15,12 @@ angular.module('syncthing.core')
                                 ctrl.$setValidity('validDeviceid', true);
                             }
                         });
+                        //Prevents user from adding a duplicate ID
+                        var matches = scope.devices.filter(function (n) {
+                            return n.deviceID == viewValue;
+                        }).length;
+                        if (matches > 0) ctrl.$setValidity('unique', false);
+                        else ctrl.$setValidity('unique', true);
```

Comment:
Nearly all of the JavaScript code for this project has conditional blocks on their own line and surrounded by brackets, as lines 12-16 of this file. I recommend following the local custom.